app/src/main/java/app/dto/servicedefinition/UpdateServiceDefinitionAttributeDTO.java

```diff
@@ -12,53 +12,56 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package app.model.service.servicedefinition;
+package app.dto.servicedefinition;
 
+import app.model.service.AttributeDataType;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlElementWrapper;
 import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
 import io.micronaut.core.annotation.Introspected;
 
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotBlank;
 import java.util.List;
 
 @Introspected
-public class ServiceDefinitionAttribute {
+public class UpdateServiceDefinitionAttributeDTO {
 
     private String code;
-    private boolean variable;
     private AttributeDataType datatype;
-    private boolean required;
+    private Boolean variable;
+    private Boolean required;
     private String description;
 
+    @Min(1)
```

Comment:
I think it should be `@Min(0)` so that it corresponds to the position that the element occupies in the list.
That way in the future, if we wanted to allow changing the attribute order with drag and drop (in the UI), we can use the new indexes in the list to derive the new attribute orders.